### PR TITLE
Update Switch component with new theme color variables

### PR DESCRIPTION
**Pull request by pastelcyborg**
## Done

- Updated colors in Switch component to use new theme variables

Fixes [WD-11873](https://warthogs.atlassian.net/browse/WD-11873)

## QA

- Open the [Switch example](https://vanilla-framework-5185.demos.haus/docs/examples/patterns/switch?theme=light)
- Switch color theme
- Ensure all UI elements look as expected throughout every interaction

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [X] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [X] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix release (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [X] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical/vanilla-framework/blob/main/releases.yml).

[WD-11873]: https://warthogs.atlassian.net/browse/WD-11873?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: scss/_patterns_switch.scss [hunk 1]
```diff
@@ -35,7 +35,7 @@ $knob-size: $sp-unit * 2;
   }
 
   .p-switch__slider {
-    background: $color-mid-light;
+    background: $colors--theme--background-neutral-default;
     border-radius: $knob-size;
     box-shadow: inset 0 2px 5px 0 transparentize($color-dark, 0.8);
```
Comment (jmuzina) [inline scss/_patterns_switch.scss:40]: @bartaz Will leaving this as `$color-dark` (legacy generic color variable) cause any issues with the tokens work that we will be doing soon?

**Comment by jmuzina** [comment]
I'm noticing that the labeled Percy job is not running, even though it ran in the QA I ran when testing that label workflow. It's possible either Github changed something with how `contains` works or that something about this PR is different than the ones that I tested. Investigating it now.

File: scss/_patterns_switch.scss (hunk 1 above)
Comment (lyubomir-popov) [reply scss/_patterns_switch.scss:40]: what si background-neutral? $color-dark should be deprecated and replaced by $color-x-dark (pure black) I think.

**Comment by lyubomir-popov** [comment]
Looks good! Couple of nit picks:

> * We don't have a themed `focus` color for links, so I left this the default `focus` color
- can we please use pure black for all shadows (`$color-x-dark`)
- can we please use #9cf for focus in the dark theme
image

**Comment by bartaz** [comment]
> Cool, thanks @lyubomir-popov! The shadow update is trivial, but for the focus color, I'll probably need to create a new theme color. Any thoughts about this, @bartaz?

If we need one, than we need one. For now I assumed focus colour has good enough contrast regardless of the theme.
But we can surely add one, `$colors--theme--color-focus` I guess, (and consistently the CSS variable), with the values that are needed.

And yes, as you mentioned in MM, there is the `%vf-focus` that is used across the board that would need updating as well (possibly separately).

Maybe it would actually be worth doing a separate PR for focus colour theming, and only then update this one?

**Comment by jmuzina** [comment]
Blocked pending #5192

**Comment by bartaz** [comment]
Code looks good, but let's rebase it now, that the percy combined examples have been merged and see how that runs.

**Review by bartaz — APPROVED** [review]
Good to go!

**Pull request merged**